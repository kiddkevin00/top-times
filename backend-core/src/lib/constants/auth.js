exports.ERROR_MSG = {
  JWT_INVALID: 'The provided JWT is invalid.',
  EMAIL_ALREADY_SIGNUP: 'This email address has already been taken.',
  LOGIN_INFO_INCORRECT: 'Incorrect login information.',
  USER_NOT_FOUND: 'The provided user is no longer exited in database.',
  TIMEZONE_NOT_FOUND: 'The provided timezone is no longer exited in database.',
  PERMISSION_DENIED: 'You are not authorized to perform this operation.',
};

exports.CORS = {
  WHITELIST: [
    'https://www.toptimes.com',
    'http://0.0.0.0:8088',
    'http://127.0.0.1:8088',
    'http://localhost:8088',
  ],
};

exports.ROLES = {
  MEMBER: 'Member',
  MANAGER: 'Manager',
  ADMIN: 'Admin',
};

exports.SALT_ROUNDS = 5;
