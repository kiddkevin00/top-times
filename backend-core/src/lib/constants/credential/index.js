const prodCredential = require('./production');
const testCredential = require('./test');
const devCredential = require('./development');

const env = process.env.NODE_ENV;
let credential;

switch (env) {
  case 'production':
    credential = prodCredential;
    break;
  case 'test':
    credential = testCredential;
    break;
  default:
    credential = devCredential;
    break;
}

module.exports = exports = credential;
