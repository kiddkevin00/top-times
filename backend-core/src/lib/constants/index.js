const systemConstants = require('./system');
const storeConstants = require('./store');
const authConstants = require('./auth');
const credentialConstants = require('./credential/');

module.exports = exports = {
  SYSTEM: systemConstants,
  STORE: storeConstants,
  AUTH: authConstants,
  CREDENTIAL: credentialConstants,
};
