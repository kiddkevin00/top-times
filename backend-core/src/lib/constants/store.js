exports.ERROR_MSG = {
  STORAGE_TYPE_NOT_FOUND: 'The requesting storage type is not found.',
  INTERFACE_NOT_IMPLEMENTED: 'The implementation for the requested interface is not found.',
};

exports.TYPES = {
  MONGO_DB: 'mongo-store',
  POSTGRES: 'postgres-store',
};

exports.OPERATIONS = {
  INSERT: 'insert',
  SELECT: 'select',
  UPDATE: 'update',
  DELETE: 'delete',
  UPSERT: 'upsert',
};

exports.TABLE_NAMES = {
  USER: 'user',
  TIME_ZONE: 'timeZone',
};
