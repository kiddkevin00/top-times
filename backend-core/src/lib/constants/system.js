exports.HTTP_STATUS_CODES = {
  OK: 200,
  CREATED: 201,
  NO_CONTENT: 204,
  TEMPORARY_REDIRECT: 307,
  PERMANENT_REDIRECT: 308,
  BAD_REQUEST: 400,
  UNAUTHENTICATED: 401,
  FORBIDDEN: 403,
  NOT_FOUND: 404,
  INTERNAL_SERVER_ERROR: 500,
  NOT_IMPLEMENTED: 501,
};
