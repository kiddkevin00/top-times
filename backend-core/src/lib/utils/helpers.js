const uuid = require('uuid');

exports.generateUniqueId = () => uuid.v4();
