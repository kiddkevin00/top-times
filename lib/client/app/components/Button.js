import React, { Component } from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';

import '../../static/assets/styles/components/button.scss';

// eslint-disable-next-line react/prefer-stateless-function
export default class Button extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    className: PropTypes.string,
    onClick: PropTypes.func.isRequired,
    type: PropTypes.string,
    disabled: PropTypes.bool,
  };

  static defaultProps = {
    className: '',
    type: 'button',
    disabled: false,
  };

  render() {
    const className = cx('pn-button', this.props.className);

    return (
      <button
        className={className}
        onClick={this.props.onClick}
        type={this.props.type}
        disabled={this.props.disabled}
      >
        {this.props.children}
      </button>
    );
  }
}
