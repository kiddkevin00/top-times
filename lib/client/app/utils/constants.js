import moment from 'moment-timezone';

const noop = () => {};

const errorMessages = {
  unauthenticated: 'unauthenticated',
};

const jwtStorageKey = 'jwt';

const timeZonesSortBy = {
  name: 'Sort By Name',
  city: 'Sort By City',
  differenceWithGmt: 'Sort By Difference with GMT',
};

const usersSortBy = {
  name: 'Sort By Name',
  role: 'Sort By Role',
  timeZone: 'Sort By Timezone',
};

const roles = {
  member: 'Member',
  manager: 'Manager',
  admin: 'Admin',
};

const allAvailableTimeZones = moment.tz.names().map(timeZone => timeZone.replace(/_/g, ' '));

export {
  noop,
  errorMessages,
  jwtStorageKey,
  timeZonesSortBy,
  usersSortBy,
  roles,
  allAvailableTimeZones,
};
