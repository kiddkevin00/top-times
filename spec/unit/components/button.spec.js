import Button from '../../../lib/client/app/components/Button';
import { noop } from '../../../lib/client/app/utils/constants';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

describe('Button component', () => {
  let props;

  beforeEach(() => {
    props = {
      onClick: noop,
      className: 'test-class',
    };
  });

  it('should always have predefined CSS classes', () => {
    const wrapper = shallow(<Button {...props}>click me</Button>);

    expect(wrapper.find('.pn-button').length).toBe(1);
  });

  describe('when children is not a string', () => {
    it('should be rendered correctly', () => {
      const wrapper = shallow(<Button {...props}><span>click me</span></Button>);

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('triggers `onClick` handler when clicking', () => {
      props.children = <span>click me</span>;
      props.onClick = jest.fn();
      const wrapper = shallow(<Button {...props} />);

      wrapper.find('button').simulate('click');

      expect(props.onClick).toHaveBeenCalledTimes(1);
    });
  });

  describe('when children is a string', () => {
    it('should be rendered correctly', () => {
      const wrapper = shallow(<Button {...props}>click me</Button>);

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should match the same string with the native `button` DOM element', () => {
      const buttonText = 'click me';

      props.children = buttonText;
      const wrapper = shallow(<Button {...props} />);

      expect(wrapper.find('button').props().children).toBe(buttonText);
    });

    it('should have `onClick()` handler passed into native `button` DOM element', () => {
      props.children = 'click me';
      props.onClick = () => {};
      const wrapper = shallow(<Button {...props} />);

      expect(wrapper.find('button').props().onClick).toBe(props.onClick);
    });
  });
});
