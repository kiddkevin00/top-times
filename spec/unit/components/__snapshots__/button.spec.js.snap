// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component when children is a string should be rendered correctly 1`] = `
<button
  className="pn-button test-class"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  click me
</button>
`;

exports[`Button component when children is not a string should be rendered correctly 1`] = `
<button
  className="pn-button test-class"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <span>
    click me
  </span>
</button>
`;
